app/com/yetu/oauth2resource/services/tokenvalidation/OAuth2TokenValidationService.scala

```diff
@@ -81,21 +82,43 @@ class OAuth2TokenValidationService(oAuth2ProviderSettings: OAuth2ProviderSetting
       val json = Json.parse(jWSObject.getPayload.toString)
       val res = json.as[ValidationResponse]
 
-      def ifEmpty: Either[Exception, ValidationResponse] = {
-        val ex: ValidationTokenException = ValidationTokenException("Token is corrupted on expiration data provided")
-        Logger.error("Validation exception, expired data is not provided", ex)
-        Left(ex)
-      }
-
-      res.exp.fold(ifEmpty)(time =>
-        if (time >= System.currentTimeMillis / 1000) {
-          Logger.debug(s"Token validated with key and time, exp date is $res")
-          Right(res)
-        } else {
-          Logger.error(s"Token validation failed due to expiration ex time is $res")
-          Left(new TokenExpiredException(s"Expired access token with $res"))
-        })
+      verifyExp(res).fold(Left(_), r => verifyAudience(r))
+    }
+  }
+
+  def verifyExp(jwt: ValidationResponse): Either[Exception, ValidationResponse] = {
+    def ifEmpty: Either[Exception, ValidationResponse] = {
+      val ex: ValidationTokenException = ValidationTokenException("Token is corrupted on expiration data provided")
+      Logger.error("Validation exception, expired data is not provided", ex)
+      Left(ex)
     }
+
+    jwt.exp.fold(ifEmpty)(time =>
+      if (time * 1000 >= System.currentTimeMillis) {
+        Logger.debug(s"Token validated with key and time, exp date is $jwt")
+        Right(jwt)
+      } else {
+        Logger.error(s"Token validation failed due to expiration ex time is $jwt")
+        Left(new TokenExpiredException(s"Expired access token with $jwt"))
+      })
+  }
+
+  def verifyAudience(jwt: ValidationResponse) = {
+    def ifNoAudience: Either[Exception, ValidationResponse] = {
+      val ex = InvalidAudienceException("Token is corrupted on expiration data provided")
+      Logger.error("Validation exception, expired data is not provided", ex)
+      Left(ex)
+    }
+
+    jwt.aud.fold(ifNoAudience)(aud =>
+      Defaults.targetAudience.intersect(aud.split(" ").toList) match {
```

Comment:
There is a possibility to pass audience list as config parameter. 
Default object represents hash of values, taken from params.